### PR TITLE
feat(router): change temp locker config as enable only

### PR DESCRIPTION
**Pull request by sai-harsha-vardhan**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
change temp locker config as enable only, temp locker would be utilized only for specific connector <> payment_method which is enabled through env

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Tested Manually
1. Enable temp locker for stripe cards
image
image

2. Remove cards payment_method from config
image
image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by ArjunKarthik — APPROVED** [review]

File: crates/router/src/core/payments/operations/payment_complete_authorize.rs [hunk 1]
```diff
@@ -300,11 +300,6 @@ impl<F: Clone + Send, Ctx: PaymentMethodRetrieve> Domain<F, api::PaymentsRequest
     )> {
         let (op, payment_method_data) =
             helpers::make_pm_data(Box::new(self), state, payment_data).await?;
-
-        utils::when(payment_method_data.is_none(), || {
-            Err(errors::ApiErrorResponse::PaymentMethodNotFound)
-        })?;
```
Comment (Narayanbhat166) [inline crates/router/src/core/payments/operations/payment_complete_authorize.rs:306]: why is this removed?

**Review by Narayanbhat166 — APPROVED** [review]
